test/application.js

```diff
@@ -412,9 +412,12 @@ describe('app.respond', function(){
         this.status = 200;
         res.setHeader("Content-Type", "text/html")
         res.write('Hello');
-        setTimeout(function(){
-          res.end("Goodbye")
-        }, 0);
+        yield new Promise(function(resolve){
+          setTimeout(function(){
+            res.end("Goodbye");
+            resolve();
+          }, 0);
+        });
```

Comment:
> the previous version should still work.

i mean the functionality shouldn't change, though i see how it gets complicated. this might be a breaking change, then